### PR TITLE
MB-14922 Extract out the application setup section

### PR DESCRIPTION
**Pull request by YanZ777**
[MB-14922](https://dp3.atlassian.net/browse/MB-14922)
Took out most of the items so that the linking all works.
The rest of the application setup will be handled in a separate PR.
I would prefer to avoid overhauling the setup instructions in this PR, but I wanted to make sure I got the general organization correct.

## Things I'd like for reviewers to specifically look out for
Are the heading numbers correct? e.g. h1, h2, etc. 
Are the links correct? 
Is the organization of the files good?

## Questions for reviewers  
Should I preclude the file headers with "Setup:"? I took it out from the original README because they're all in their own files now under the "Application Setup" tab.

[MB-14922]: https://dp3.atlassian.net/browse/MB-14922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/about/application-setup/setup-using-nix/add-or-update-packages.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Add or Update Packages
```
Comment (rogeruiz) [inline docs/about/application-setup/setup-using-nix/add-or-update-packages.md:1]:
```suggestion
---
sidebar_position: 4
---
# Add or Update Packages
```

File: docs/about/application-setup/setup-using-nix/clean-up-local-environment.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Clean Up Local Environment
```
Comment (rogeruiz) [inline docs/about/application-setup/setup-using-nix/clean-up-local-environment.md:1]:
```suggestion
---
sidebar_position: 3
---

# Clean Up Local Environment
```

File: docs/about/application-setup/setup-using-nix/initial-setup.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Nix Initial Setup
```
Comment (rogeruiz) [inline docs/about/application-setup/setup-using-nix/initial-setup.md:1]:
```suggestion
---
sidebar_position: 1
---

# Nix Initial Setup
```

File: docs/about/application-setup/setup-using-nix/install-dependencies.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Install Dependencies
```
Comment (rogeruiz) [inline docs/about/application-setup/setup-using-nix/install-dependencies.md:1]:
```suggestion
---
sidebar_position: 2
---

# Install Dependencies
```

File: docs/about/application-setup/setup-using-nix/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+id: Setup Using Nix
+sidebar_position: 1
```
Comment (rogeruiz) [inline docs/about/application-setup/setup-using-nix/README.md:3]:
```suggestion
sidebar_position: 3
```

Let's be explicit that the Nix guide is after the Manual guide here. The should not both have a `sidebar_position` of 1.

File: docs/about/application-setup/setup-manually/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+id: Setup Manually
+sidebar_position: 1
```
Comment (rogeruiz) [inline docs/about/application-setup/setup-manually/README.md:3]:
```suggestion
sidebar_position: 2
```

**Review by rogeruiz — CHANGES_REQUESTED** [review]
❌ This PR needs changes ...

I'm requesting changes, but this is still very good @YanZ777. My changes are mostly on the order of the items in the Sidebar for this section. The Nix documents needed to be ordered in the correct steps. And the overall Application Setup section should be ordered from `base-setup` to `setup-manually` and then `setup-using-nix` being the last one. 

> Are the heading numbers correct? e.g. h1, h2, etc.

Yes the heading numbers look good for me. I'm looking at the files in `docs/about/application-setup/*.md` and matching them against the old order from `docs/about/Welcome.md`. Some of the descendent headings are incorrect, but I'm providing links to suggestions you can apply in GitHub or locally.

> Are the links correct?

None of the internal links are broken, after building the application.

> Is the organization of the files good?

Organization looks great. Awesome work on this.

**Review by kctruss — APPROVED** [review]
My gosh, it's beautiful! 😭 😍

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

**Pull request merged**